### ISSUE TITLE
Add support for collapse param in Search

### ISSUE THREAD

**Issue by BANOnotIT** [elastic/elasticsearch-dsl-py#1215]
There's no method for collapse feature. [Collapsing feature](https://www.elastic.co/guide/en/elasticsearch/reference/7.2/search-request-collapse.html) is vary usefull for building suggestions.

To use this feature right now in v7.0.0 of *e-dsl* I can use this workaround:
```python
s = Search(...)
s = s.update_from_dict({'collapse': {'field': 'some.field'}})
```

It'll be much easier to use it in a way like that:
```python
s = Search(...)
s = s.collapse('some.field', 'some.another.field')
s = s.collapse('some.complicated.field'={
  'inner_hits': ...,
  '...': ...
})
````

**Comment by nirLumigo** [comment]
What is the status of the feature?

**Comment by AhmedNourJamalElDin** [comment]
Any update?

**Comment by ZeJ0hn** [comment]
Any news ?

**Comment by DDusky** [comment]
You save me !!! thanks

**Comment by nuhour** [comment]
could someone handle this issue?pls....@sethmlarson @dsanders11 @mattjmcnaughton

**Comment by BrowenChen** [comment]
any update?

**Comment by pquentin** [comment]
This was fixed in #1649 (thanks @qcoumes!) and is part of the 8.11.0 release. Thank you for your patience.
